### ISSUE TITLE
@Value.Lazy is ignored for methods with parameters

### ISSUE THREAD

**Issue by dalippa** [immutables/immutables#445]
If I have 2 methods in a class:
```java
@Value.Immutable
class Foo {
  @Value.Lazy
  public Instant foo() {
    return Instant.now();
  }

  @Value.Lazy
  public Instant bar(int param) {
    return Instant.now();
  }
  
}
```

In this case, `Foo.foo()` is cached, but `Foo.bar()` is not. I would expect one of the following to occur:
* Compiler error/warning showing that `bar()` cannot be cached.
* A parameter to `@Value.Lazy` that provides uses a caching mechanism (parameter tuple -> value) with eviction protocols to avoid memory growth exponential with the number of parameters/their range.

**Comment by elucash** [comment]
You're right, there should be better a compile error or warning. We are not planning to support parameters with far reaching consequences, it should be implemented explicitly in code
